x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/RegexExtract.java

```diff
@@ -40,10 +41,19 @@ public Expression input() {
         return input;
     }
 
+    /**
+     * Upon parsing, these are named according to the {@link Dissect} or {@link Grok} pattern, but can be renamed without changing the
+     * pattern.
+     */
     public List<Attribute> extractedFields() {
         return extractedFields;
     }
 
+    @Override
+    public List<Attribute> generatedAttributes() {
```

Comment:
What is the difference between `generatedAttributes` and `extractedFields`? Why not calling `extractedFields()` directly?

Reply:
`extractedFields()` exists on Grok and Dissect, but not on `Eval` nor `Enrich`; these have `fields()` (`Alias`es, though) and `enrichFields()` (`NamedExpression`s).

Having all of them implement a common interface with this `generatedAttributes` makes it much easier to write the pushdown rule for Grok, Dissect, Eval and Enrich - and also to test them.

Additionally, we're in the process of adding more plan nodes that, with respect to shadowing, should behave the same: `Lookup` and `Inlinestats`. Having an interface should make the rules easier to reason about.